fossunited/fossunited/doctype/foss_event_schedule/foss_event_schedule.py

```diff
@@ -5,4 +5,25 @@
 
 
 class FOSSEventSchedule(Document):
+    # begin: auto-generated types
+    # This code is auto-generated. Do not modify anything in this block.
+
+    from typing import TYPE_CHECKING
+
+    if TYPE_CHECKING:
+        from frappe.types import DF
+
+        category: DF.Literal["Talk", "Lightning Talk", "Workshop", "Panel Discussion", "Opening Note", "Break", "Other"]
+        end_time: DF.Time | None
+        hall: DF.Data | None
+        linked_cfp: DF.Link | None
+        other_category: DF.Data | None
+        parent: DF.Data
+        parentfield: DF.Data
+        parenttype: DF.Data
```

Comment:
`parent` `parentfield` and `parenttype` are set by frappe since this doctype is a child doctype. Suppose I have an event with "ChennaiFOSS" id and it is a doctype of "Event" type.

Event Doctype has a field `schedules` which is of type "Table" (links to child doctypes).

Then whenever i create a schedule entry for the event (as shown in above image), the fields will be mapped as such:

```py
{
    parent: 'ChennaiFOSS'
    parentfield: 'schedules',
    parenttype: 'Event'
}
```

Reply:
Like I asked above, if `Event` is the parent doctype, why does the `FossEventSchedule` Python class inherit from `Document`?